jsapp/js/components/modalForms/encryptForm.es6

```diff
@@ -0,0 +1,157 @@
+    };
+
+    autoBind(this);
+  }
+
+  componentDidMount() {
+    this.listenTo(stores.asset, this.onAssetsChange);
+
+    if (!this.state.asset && this.state.assetUid) {
+      if (stores.asset.data[this.state.assetUid]) {
+        this.onAssetChange(stores.asset.data[this.state.assetUid]);
+      } else {
+        stores.allAssets.whenLoaded(this.props.assetUid, this.onAssetChange);
+      }
+    }
+  }
+
+  onAssetChange(asset) {
+    this.setState({
+      asset: asset,
+      submissionURL: asset.content.settings.submission_url,
+      publicKey: asset.content.settings.public_key
+    });
+
+    stores.pageState.showModal({
+      type: MODAL_TYPES.ENCRYPT_FORM,
+      asset: asset
+    });
+  }
+
+  onAssetsChange(assetsList) {
+    let uid;
+    if (this.state.asset) {
+      uid = this.state.asset.uid;
+    } else if (this.state.assetUid) {
+      uid = this.state.assetUid;
+    }
+    this.onAssetChange(assetsList[uid]);
+  }
+
+  updateAsset(content) {
+    actions.resources.updateAsset(
+      this.props.asset.uid,
+      {content: JSON.stringify(content)}
+    );
+  }
+  onSubmit(evt) {
+    evt.preventDefault();
+
+    var content = this.state.asset.content;
+    content.settings.submission_url = this.state.submissionURL;
+    content.settings.public_key = this.state.publicKey;
+    this.updateAsset(content);
+  }
+  onRemove(evt) {
+    evt.preventDefault();
+    this.setState({clearEncryption: true});
+
+    var content = this.state.asset.content
+    content.settings.submission_url = '';
+    content.settings.public_key = '';
+    this.updateAsset(content);
+  }
+
+  onSubmissionURLChange (newSubmissionURL) {
+    this.setState({
+      submissionURL: newSubmissionURL,
+      clearEncryption: false
+    });
+  }
+  onPublicKeyChange (newPublicKey) {
+    this.setState({
+      publicKey: newPublicKey,
+      clearEncryption: false
+    });
+  }
+
+  openEncryptionHelp() {
+    window.open('https://support.kobotoolbox.org/encrypting_forms.html', '_blank');
+  }
+
+  render () {
+    return (
+      <bem.FormView__form m='add-language-fields'>
+        <bem.FormView__cell m='encrypt-url'>
+          <bem.FormModal__item>
+            <label>{t('Submission URL')}</label>
+            <TextBox
+              value={!this.state.clearEncryption ? this.state.submissionURL : ''}
+              onChange={this.onSubmissionURLChange}
+            />
+          </bem.FormModal__item>
+        </bem.FormView__cell>
+
+        <bem.FormView__cell m='encrypt-key'>
+          <bem.FormModal__item>
+            <label>{t('Public key')}</label>
+            <TextBox
+              value={!this.state.clearEncryption ? this.state.publicKey : ''}
+              onChange={this.onPublicKeyChange}
+            />
+          </bem.FormModal__item>
+        </bem.FormView__cell>
+
+        <bem.FormView__cell m='submit-button'>
+          <button
+            className='mdl-button mdl-js-button mdl-button--raised mdl-button--colored'
```

Comment:
`mdl-js-button` doesn't do anything, please remove it :)

Reply:
neither does `mdl-button--danger` anymore :D

Reply:
oh right, it's `mdl-button--red` now :)